src/pipecat/services/speechmatics/stt.py

```diff
@@ -412,63 +662,61 @@ async def _disconnect(self) -> None:
 
-    def _process_config(self, transcription_config: Optional[TranscriptionConfig] = None) -> None:
+    def _process_config(self) -> None:
         """Create a formatted STT transcription config.
 
-        This takes an optional TranscriptionConfig object and populates it with the
-        values from the STT service. Individual parameters take priority over those
-        within the config object.
-
-        Args:
-            transcription_config: Optional transcription config to use.
+        Creates a transcription config object based on the service parameters. Aligns
+        with the Speechmatics RT API transcription config.
         """
         # Transcription config
-        if not transcription_config:
-            transcription_config = TranscriptionConfig(
-                language=self._language_code or "en",
-                domain=self._domain,
-                output_locale=self._output_locale_code,
-                operating_point=self._operating_point,
-                diarization="speaker" if self._enable_speaker_diarization else None,
-                enable_partials=self._enable_partials,
-                max_delay=self._max_delay or 2.0,
-            )
-        else:
-            if self._language_code:
-                transcription_config.language = self._language_code
-            if self._domain:
-                transcription_config.domain = self._domain
-            if self._output_locale_code:
-                transcription_config.output_locale = self._output_locale_code
-            if self._operating_point:
-                transcription_config.operating_point = self._operating_point
-            if self._enable_speaker_diarization:
-                transcription_config.diarization = "speaker"
-            if self._enable_partials:
-                transcription_config.enable_partials = self._enable_partials
-            if self._max_delay:
-                transcription_config.max_delay = self._max_delay
+        transcription_config = TranscriptionConfig(
+            language=self._language,
+            domain=self._domain,
+            output_locale=self._output_locale,
+            operating_point=self._operating_point,
+            diarization="speaker" if self._enable_diarization else None,
+            enable_partials=self._enable_partials,
+            max_delay=self._max_delay,
+        )
 
-        # Diarization
-        if self._enable_speaker_diarization and self._max_speakers:
-            transcription_config.speaker_diarization_config = SpeakerDiarizationConfig(
-                max_speakers=self._max_speakers,
-            )
+        # Additional vocab
+        if self._additional_vocab:
+            transcription_config.additional_vocab = [
+                {
+                    "content": e.content,
+                    "sounds_like": e.sounds_like,
+                }
+                for e in self._additional_vocab
+            ]
```

Comment:
I'm getting a warning in my IDE:
`Expected type 'dict | None', got 'list[dict[str, str | list[str]]]' instead.`

Just wanted to confirm if it is this the expected type, or should we be sending a single dictionary instead ?